plugin-server/src/cdp/hog-function-manager.ts

```diff
@@ -89,66 +127,77 @@ export class HogFunctionManager {
         }
     }
 
-    public fetchHogFunction(id: HogFunctionType['id']): Promise<HogFunctionType | null> {
-        return fetchHogFunction(this.postgres, id)
+    public async fetchHogFunction(id: HogFunctionType['id']): Promise<HogFunctionType | null> {
+        const items: HogFunctionType[] = (
+            await this.postgres.query(
+                PostgresUse.COMMON_READ,
+                `SELECT ${HOG_FUNCTION_FIELDS.join(', ')}
+                FROM posthog_hogfunction
+                WHERE id = $1 AND deleted = FALSE`,
```

Comment:
Do we want to add `and team_id = $2` here for safety when fetching by `id`?

Reply:
Hmmm maybe. The auth is delegated to the main service so we don't really need to do any explicit checking here but I might add to be extra safe

Reply:
Actually this is covered by the API handler